### PR TITLE
config: switch eslint errors to warning to suppress CI failures

### PR DESCRIPTION
**Pull request by Roboromeo1**
*Any italic text should be deleted from the final Pull Request text, including this line*

# Description

*Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.*

Fixes # (issue)

## Type of change

*Please delete options that are not relevant.*

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

*Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration*

## Testing Checklist:

- [ ] Tested in latest Chrome
- [ ] Tested in latest Safari
- [ ] Tested in latest Firefox

# Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have requested a review from @macite and @jakerenzella on the Pull Request

### CODE REVIEW

File: .eslintrc.json [hunk 1]
```diff
@@ -9,18 +9,21 @@
       },
       "extends": ["plugin:@angular-eslint/recommended"],
       "rules": {
+        "@angular-eslint/no-empty-lifecycle-method": "warn",
+        "@angular-eslint/component-class-suffix":"warn",
+        "@angular-eslint/no-output-on-prefix":"warn",
         "@angular-eslint/directive-selector": [
           "error",
```
Comment (macite) [inline .eslintrc.json:16]: Lets not remove these "errors" all together. Instead, can we change them to "warn"? Then we can create a new PR to fix all of the warnings... followed by turning this back to errors 😄

**Review by macite — CHANGES_REQUESTED** [review]
Please change strategy to solve issue - moving errors to warnings so that they are still raised and can be addressed.

**Comment by macite** [comment]
Not sure I would call this a new "feature" PR. Quality or config is probably more accurate.

**Comment by jakerenzella** [comment]
Further, it seems you have not built the project with the correct version of node, which has resulted in the large lock file. Please ensure Node v14 is used to build Angular projects such as Doubtfire

**Comment by jakerenzella** [comment]
> Not sure I would call this a new "feature" PR. Quality or config is probably more accurate.

Also not sure why the context submessage has been used

File: .eslintrc.json [hunk 2]
```diff
@@ -12,22 +12,24 @@
...
-            // "type": "element",
-            "prefix": "df",
-            "style": "kebab-case"
-          }
-        ],
+        "@angular-eslint/component-selector":"warn",
+        "@angular-eslint/directive-selector":"warn",
+        // "@angular-eslint/directive-selector": [
+        //   "error",
+        //   {
```
Comment (macite) [inline .eslintrc.json:19]: Can we keep these details but switch to warn by updating the above "error" to "warn"?

File: .eslintrc.json [hunk 3]
```diff
@@ -12,22 +12,24 @@
...
+        //     "type": "attribute",
+        //     "prefix": "df",
+        //     "style": "camelCase"
+        //   }
+        // ],
+        // "@angular-eslint/component-selector": [
+        //   "error",
```
Comment (macite) [inline .eslintrc.json:26]: Same here

**Review by macite — CHANGES_REQUESTED** [review]
Can you update to keep the existing settings but switch them to warn.

**Review by macite — APPROVED** [review]

**Comment by jakerenzella** [comment]
@macite This shouldn't have been merged since it needed to be built with Node v14. This caused the Lock file to be bloated.

**Comment by macite** [comment]
Sorry I didn't notice the lock file change 😞 @jakerenzella

**Comment by Roboromeo1** [comment]
@macite @jakerenzella sorry for the inconvenience caused I apologise got that. Should I change the node version to 14 and create a new Pr ? 😅

**Comment by jakerenzella** [comment]
> @macite @jakerenzella sorry for the inconvenience caused I apologise got that. Should I change the node version to 14 and create a new Pr ? 😅

No worries at all it isn't well documented so not your fault. This is fine I'll look at this in another pr. Thanks for your contribution.

**Comment by Roboromeo1** [comment]
Thank you , I'll fix it asap and commit with my next pr 🙌

**Comment by jakerenzella** [comment]
Actually what might help if you have some availability for this @Roboromeo1 is if you would be able to add some documentation to the web repo that Node v14 needs to be used when installing/updating packages for doubtfire-web!

**Pull request merged**